frame/support/procedural/src/construct_runtime/parse.rs

```diff
@@ -198,6 +199,84 @@ impl Parse for PalletDeclaration {
 	}
 }
 
+/// A struct representing a path to a pallet. `PalletPath` is almost identical to the standard
+/// Rust path with a few restrictions:
+/// - No leading colons allowed
+/// - Path segments can only consist of identifers; angle-bracketed or parenthesized segments will
+///   result in a parsing error (except when specifying instances)
+#[derive(Debug, Clone)]
+pub struct PalletPath {
+	pub inner: Path,
+}
+
+impl Parse for PalletPath {
+	fn parse(input: ParseStream) -> Result<Self> {
+		let mut lookahead = input.lookahead1();
+		let mut segments = Punctuated::new();
+
+		if lookahead.peek(Token![crate])
+			|| lookahead.peek(Token![self])
+			|| lookahead.peek(Token![super])
+			|| lookahead.peek(Ident)
```

Comment:
a path could also start with `::`, I wonder if wouldn't be simpler to parse for `syn::Path` and then for `Option<syn::MethodTurbofish>` and then check on the correctness of the items.
Anyway the current implementation is ok to me.

Reply:
The problem is that it would definitely consume the `<Instance>` and `{PalletPart}` segments, and you can't really tell it to stop parsing once it sees a brace or an angle bracket. I might be wrong on this though, but it's definitely something we can look into in a follow-up.